components/board/digital_interrupts.go

```diff
@@ -204,7 +210,7 @@ func (i *ServoDigitalInterrupt) Value(ctx context.Context, extra map[string]inte
 
 // Tick records the time between two successive low signals (pulse width). How it is
 // interpreted is based off the consumer of Value.
-func (i *ServoDigitalInterrupt) Tick(ctx context.Context, high bool, now uint64) error {
+func (i *ServoDigitalInterrupt) Tick(ctx context.Context, high bool, now uint32) error {
 	diff := now - i.last
```

Comment:
2^32 microseconds is about 72 minutes. Assuming `now` and `i.last` are in microseconds, what happens if you run your robot for 2 hours? Won't there be a changeover in the middle where `diff` is large and negative?

I think to be on the safe side we should go back to 64-bit integers, though I'm open to disagreement.

edit: :astonished: No, assuming `diff` is also a uint32, it'll probably overflow in the usual way and do the right thing here! The only worry is if `now` and `i.last` occurred more than 72 minutes apart, which is not a problem for the ultrasonic sensor but might be a problem for other interrupt-related things.

Reply:
Make sense should coerce diff as a var uint32